File: pkg_zero/src/add_two_ints_client.cpp
#include "ros/ros.h"
#include "pkg_zero/AddTwoInts.h"
#include <cstdlib>

int main(int argc, char **argv)
{
    ros::init(argc, argv, "add_two_ints_client");
    if (argc != 3)
    {
        ROS_ERROR("2 arguments are required!!");
        return 1;
    }

    ros::NodeHandle n;
    ros::ServiceClient client = n.serviceClient<pkg_zero::AddTwoInts>("add_two_ints");

    pkg_zero::AddTwoInts srv;
    srv.request.a = atoll(argv[1]);
    srv.request.b = atoll(argv[2]);
    if (client.call(srv))
    {
        ROS_INFO("sum: %ld", (long int)srv.response.sum);
    }
    else
    {
        ROS_ERROR("failed to call the service add_two_ints");
        return 1;
    }

    return 0;
}

File: pkg_zero/scripts/add_two_ints_client.py
#!/usr/bin/env python3
import sys
import rospy
from pkg_zero.srv import AddTwoInts

def add_two_ints_client(x, y):
    rospy.wait_for_service('add_two_ints')
    try:
        handle = rospy.ServiceProxy('add_two_ints', AddTwoInts)
        res = handle(x, y)
        return res.sum
    except rospy.ServiceException as e:
        rospy.loginfo(f"service call failed: {e}")

if __name__ == '__main__':
    if len(sys.argv) == 3:
        x, y = int(sys.argv[1]), int(sys.argv[2])
    else:
        rospy.logerr("2 arguments are required!!")
        sys.exit(1)

    print(f"requesting {x} + {y}")
    result = add_two_ints_client(x, y)
    print(f"received: {x} + {y} = {result}")
